src/vs/workbench/contrib/scm/common/quickDiff.ts

```diff
@@ -0,0 +1,28 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { URI } from 'vs/base/common/uri';
+import { createDecorator } from 'vs/platform/instantiation/common/instantiation';
+import { IDisposable } from 'vs/base/common/lifecycle';
+
+export const IQuickDiffService = createDecorator<IQuickDiffService>('quickDiff');
+
+export interface QuickDiffProvider {
+	label: string;
+	rootUri: URI | undefined;
+	getOriginalResource(uri: URI): Promise<URI | null>;
+}
+
+export interface QuickDiff {
+	label: string;
+	originalResource: URI;
+}
+
+export interface IQuickDiffService {
```

Comment:
Since it is already referred to in the extension API as `QuickDiffProvider` I would like to keep referring to it as "quick diff" throughout. If you feel strongly otherwise I will change it.